writer.go

```diff
@@ -575,7 +599,9 @@ func (w *writer) run() {
 	var done bool
 	var batch = make([]Message, 0, w.batchSize)
 	var resch = make([](chan<- error), 0, w.batchSize)
+	var lstMsg *writerMessage
```

Comment:
Does this stand for `lastMsg`? Probably worth renaming because I initially read this as `listMsg`, which made it hard to understand what was happening.

Also can we not make it a pointer type? `writerMessage` is passed by value everywhere, it seems weird to have it be a pointer here.

Reply:
Yeah, `lastMsg`. I'll update the name.
I'll see what I can about removing the pointer.